packages/@aws-cdk/aws-ecs/lib/capacity-provider.ts

```diff
@@ -0,0 +1,262 @@
+import * as path from 'path';
+import { IAutoScalingGroup } from '@aws-cdk/aws-autoscaling';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import * as cr from '@aws-cdk/custom-resources';
+import { ICluster, AddCapacityOptions } from './cluster';
+import { CfnCapacityProvider } from './ecs.generated';
+
+/**
+ * Represents the CapacityProvider
+ */
+export interface ICapacityProvider extends cdk.IResource {
+  /**
+   * The name of the CapacityProvider
+   * @attribute
+   */
+  readonly capacityProviderName: string;
+}
+
+/**
+ * Options for addCapacityProvider()
+ */
+export interface CapacityProviderBase {
+  /**
+   * The name of the CapacityProvider
+   *
+   * @default - physical ID of the resource
+   */
+  readonly capacityProviderName?: string;
+
+  /**
+   * Whether to enable the managed termination protection
+   *
+   * @default True
+   */
+  readonly managedTerminationProtection?: boolean;
+
+  /**
+   * Whether to enable the managed scaling. This value will be overrided to be True
+   * if the `managedTerminationProtection` is enabled.
```

Comment:
i wonder if we should instead of overriding it to be true, we throw an error saying when `managedTerminationProtection` is enabled `managedScaling` cannot be disabled sth like that. And also have a test case for this error.

The reason is because default values for both `managedTerminationProtection` and `managedScaling` are true, and when a user set `managedScaling` to be false which means they don't want it to be enabled. Given that if we override `managedScaling` to be true then it might result in sth that users don't expect (or contradictory to their intention).